src/Tools/dotnet-monitor/CollectionRules/ExecuteAction.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal sealed class ExecuteAction : IAction<ExecuteOptions>
+    {
+        public async Task<ActionResponse> Execute(ExecuteOptions options, CancellationToken token)
+        {
+            ActionResponse executeResponse = new();
+
+            await Task.Run(() =>
+            {
+                string path = options.Path;
+                string arguments = options.Arguments;
+
+                Process process = Process.Start(path, arguments);
+
+                process.WaitForExit();
```

Comment:
I would do something like the implementation for AwaitExtensions.WaitForExitAsync, rather than blocking an entire tpl thread.

Reply:
We have test code that does this already that you can copy and adapt to this: https://github.com/dotnet/dotnet-monitor/blob/cf94ef78c80aca5f19933add98150e14d5fd6d50/src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/Runners/DotNetRunner.cs#L162

Reply:
The latest version integrates an adaptation of this - I wasn't entirely sure about how much of it was specific to the DotNetRunner file and how much was needed for our use case, so if I should have included or removed other pieces just let me know.